src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -209,7 +217,9 @@ abstract class RefChecks extends Transform {
         val (nullaries, alts) = syms.partition(sym => isNullary(sym.info))
         //assert(!alts.isEmpty)
         nullaries match {
-          case nullary :: Nil => warnDubious(nullary, syms)
+          case nullary :: Nil =>
+            if (warnable(syms))
```

Comment:
could we push the condition up to

```
if syms.lengthCompare(1) > 0 && isCompetitive(syms, ...) && warnable(syms))
```

?

Reply:
That was my first cut, but it makes the for clause too long, and it's ugly without Scala 3 to make it multiline.

I wanted to avoid indenting the block, but maybe I'll just push the conditional into the body of the for.

Probably that reads better anyway, with a complex condition.